addons/resource/tests/test_resource.py

```diff
@@ -655,3 +655,52 @@ def to_tuple(interval):
 
         work_spec = [to_tuple(x) for x in self.calendar._get_day_leave_intervals(Date.from_string('2018-04-06'), time(0), time(23, 59, 59), self.resource1_id)]
         self.assertEqual(work_spec[0], (Datetime.from_string('2018-04-06 16:00:00'), Datetime.from_string('2018-04-06 21:00:00')))
+
+
+class TestTimeType(TestResourceCommon):
+    def setUp(self):
+        super(TestTimeType, self).setUp()
+        Leave = self.env['resource.calendar.leaves']
+
+        self.other1 = Leave.create({
+            'name': 'Formation',
+            'calendar_id': self.calendar.id,
+            'resource_id': self.resource1_id,
+            'date_from': Datetime.from_string('2013-02-15 09:00:00'),
+            'date_to': Datetime.from_string('2013-02-15 19:00:00'),
+            'time_type': 'other',
+        })
+        self.other2 = Leave.create({
+            'name': 'Homeworking',
+            'calendar_id': self.calendar.id,
+            'resource_id': self.resource1_id,
+            'date_from': Datetime.from_string('2013-02-12 09:00:00'),
+            'date_to': Datetime.from_string('2013-02-12 15:00:00'),
+            'time_type': 'other',
+        })
+
+    def test_default_time_type(self):
+        self.assertEqual(self.leave1.time_type, 'leave')
+        self.assertEqual(self.leave2.time_type, 'leave')
+        self.assertEqual(self.other1.time_type, 'other')
+
+    def test_leaves_default(self):
```

Comment:
I would simplify a bit the various tests and just have 1 or 2 test using domain propagation with time_type, just to ensure the domain is effectively applied.